doc/content/reference/cli/troubleshooting.md

```diff
@@ -7,10 +7,53 @@ This section contains help for common errors encountered when using the CLI.
+
+Since your server address is most likely different than `thethings.example.com`, you will see an error about that URL not being found in your browser, the access token will not be obtained and you will not be able to execute any further CLI commands. In your terminal, you will see something like:
+
+```
+INFO Opening your browser on https://thethings.example.com/oauth/authorize?client_id=cli&redirect_uri=local-callback&response_type=code
+INFO After logging in and authorizing the CLI, we'll get an access token for future commands.
+INFO Waiting for your authorization...
+```
+
+The solution to this issue would be to replace `thethings.example.com` occurrences with the server address of your {{% tts %}} deployment.
+
+{{< note >}} You will experience the same behavior if you are using an `https` port other than `443` and you have not specified that port in the `oauth-server-address` parameter in the CLI configuration file (or with `--oauth-server-address` flag when running the CLI). For example, if you are running {{% tts %}} on `localhost`, you would have to specify the OAuth server address as `'https://localhost:8885/oauth'`.
+{{</ note >}}
+
+## Certificate Signed by Unknown Authority
+
+If the CA file path is not specified in the [CLI configuration file]({{< ref "/getting-started/cli/installing-cli#configuration" >}}) (or with `--ca` flag when running the CLI), or if you are using self-signed certificates, using `ttn-lw-cli login` might result in the following error:
+
+```
+ERROR Could not exchange OAuth access token    error=Post "https://thethings.example.com:8885/oauth/token": x509: certificate signed by unknown authority
+```
+
+In this case, the CLI configuration file should contain `ca: /path/to/ca.pem`, or you should run the CLI with `--ca="/path/to/ca.pem"` flag. If using self-signed certificates, you should also import the CA file using your web browser's certificate manager.
+
 ## No Collaborator Set
 
 Many CLI commands require that a user or organization is set to grant permissions over an entity. For example, to create an application:
 
 ```bash
 $ ttn-lw-cli applications create app1 --user-id user1
 ```
+
+## Permission Denied
+
+If a user (specified with the `--user-id` flag) does not have sufficient rights to perform the desired action, you will face an error such as:
+
+```
+error:pkg/auth/rights:insufficient_application_rights (insufficient rights for application `app1`)
+```
+
+## Invalid Identifiers
+
+When creating an entity using the CLI, if the ID value that you use does not follow {{% tts %}} naming convention, you will face an error such as:
+
+```
+error:pkg/errors:validation (invalid `gateway_id`: value does not match regex pattern "^[a-z0-9](?:[-]?[a-z0-9]){2,}$")
+```
```

Comment:
I would maybe add an example here, along with a description of the problem, e.g.

```bash
# "My_Gateway" is not a valid gateway_id because it contains upper-case letters and an underscore.
# For identifiers, use lower-case letters, numbers and dashes, e.g. "my-gateway" instead.
ttn-lw-cli gateways create My_Gateway
....
--- error:pkg/errors:validation (invalid `gateway_id`: value does not match regex pattern "^[a-z0-9](?:[-]?[a-z0-9]){2,}$")
    field=gateway_id
    name=GatewayIdentifiersValidationError
    reason=value does not match regex pattern "^[a-z0-9](?:[-]?[a-z0-9]){2,}$"
```

I would like to have an example like this (even if it seems self-explanatory or obvious) because people are reading the troubleshooting section only if the error message itself does not point them to a solution already.

Reply:
Leaving this up to you, since the target audience of the CLI is people that should be able to recognize what a 'regex does not match' error means. Yet, I believe the troubleshooting section could be an exception to the rule, especially since it is our intention for people to use the CLI for interacting with The Things Stack.